fixed-int.md

```diff
@@ -0,0 +1,44 @@
+```k
+module FIXED-INT
+    imports INT
+    imports BOOL
+
+    syntax FInt ::= "(" FInt ")"                   [bracket]
+                  | FInt ( value: Int , one: Int ) [klabel(FInt), symbol]
+ // ---------------------------------------------------------------------
+    // rule #Ceil(FInt(... value: V, one: O)) => { V >=Int 0 andBool O >Int 0 #Equals true } [anywhere, symbolic]
+
+    // Operations always produce the width of the left integer.
+    syntax FInt ::= FInt "*FInt" FInt [function]
+                  | FInt "/FInt" FInt [function]
+                  | FInt "^FInt"  Int [function]
+                  > FInt "+FInt" FInt [function]
+                  | FInt "-FInt" FInt [function]
+ // --------------------------------------------
+    rule FInt(V1, O1) *FInt FInt(V2, O2) => FInt((V1 *Int V2) /Int O2, O1)
+    rule FInt(V1, O1) /FInt FInt(V2, O2) => FInt((V1 *Int O2) /Int V2, O1) requires V2 =/=Int 0
+    rule FInt(V1, O1) ^FInt E            => FInt(V1 ^Int E, O1)
```

Comment:
This doesn't seem right, e.g.:

`(1 in width 100)^2 = FInt(100, 100) ^FInt 2 = FInt(100 ^Int 2, 100) = FInt(10000, 100) = (100 in width 100)`

but I'd expect the result to just equal `(1 in width 100)`.

A correct option could be:

`rule FInt(V1, O1) ^FInt E            => FInt((V1 ^Int E) /Int (O1 ^Int (E -Int 1)), O1) requires E >Int 0`
`rule FInt(V1, O1) ^Fint E            => FInt(O1, O1) requires E ==Int 0`

Reply:
Alright, I've corrected this and added a case for `E <Int 0` as well.

Reply:
Would appreciate you double-checking my work here though, because it can be tricky.